src/scheme.ts

```diff
@@ -11,37 +11,99 @@ import getLogger, {Logger} from './logging';
 import {Endpoint} from './discovery';
 import {retryable} from "./retries";
 import Driver from "./driver";
+import {util} from "protobufjs";
+import {OperationParams} from "./table";
 
 import SchemeServiceAPI = Ydb.Scheme.V1.SchemeService;
 import IOperationParams = Ydb.Operations.IOperationParams;
 import ListDirectoryResult = Ydb.Scheme.ListDirectoryResult;
 import DescribePathResult = Ydb.Scheme.DescribePathResult;
 import IPermissionsAction = Ydb.Scheme.IPermissionsAction;
 import IMakeDirectoryRequest = Ydb.Scheme.IMakeDirectoryRequest;
-import IPermissions = Ydb.Scheme.IPermissions;
-import {util} from "protobufjs";
 import EventEmitter = util.EventEmitter;
 
 
-function preparePermissions(action?: IPermissions | null) {
-    if (action && action.permissionNames) {
-        return {
-            ...action,
-            permissionNames: action.permissionNames.map(
-                (name) => name.startsWith('ydb.generic.') ? name : `ydb.generic.${name}`
-            )
-        };
+export class MakeDirectorySettings {
+    operationParams?: OperationParams;
+
+    withOperationParams(operationParams: OperationParams) {
+        this.operationParams = operationParams;
+        return this;
+    }
+}
+
+export class RemoveDirectorySettings {
+    operationParams?: OperationParams;
+
+    withOperationParams(operationParams: OperationParams) {
+        this.operationParams = operationParams;
+        return this;
+    }
+}
+
+export class ListDirectorySettings {
+    operationParams?: OperationParams;
+
+    withOperationParams(operationParams: OperationParams) {
+        this.operationParams = operationParams;
+        return this;
+    }
+}
+
+export class DescribePathSettings {
+    operationParams?: OperationParams;
+
+    withOperationParams(operationParams: OperationParams) {
+        this.operationParams = operationParams;
+        return this;
     }
-    return action;
 }
 
-function preparePermissionAction(action: IPermissionsAction) {
-    const {grant, revoke, set, ...rest} = action;
-    return {
-        ...rest,
-        grant: preparePermissions(grant),
-        revoke: preparePermissions(revoke),
-        set: preparePermissions(set),
+export interface Permissions {
```

Comment:
Why not IPermissions here :)?

Reply:
I revert this changes, so it's not actual now.
But the interface name will be consistent in the next changes.